src/lib/libdylink.js

```diff
@@ -937,6 +939,7 @@ var LibraryDylink = {
     var dso = {
       refcount: Infinity,
       name,
```

Comment:
So does that mean that `name` is always just the basename / so_name?  Is is name sometimes absolute too here?

Reply:
No, only the dependencies of the libraries will have only the so_name in `name`. The parent library called from `dlopen` would have absolute path in the `name`.

Reply:
> No, only the dependencies of the libraries will have only the so_name in name.

So, I think the alternative approach would be ensuring `loadDynamicLibrary` always take absolute path as an input. It requires changes here:

https://github.com/emscripten-core/emscripten/blob/38e16464cffc8f886364fe4919a712131a4c9456/src/lib/libdylink.js#L909-L917

instead of passing `dynNeeded` directly to the `loadDynamicLibrary`, we can locate the library first, and pass the absolute path.